### PR TITLE
508 pdf

### PR DESCRIPTION
**Pull request by 13bfrancis**
# Description

This is the first go round of making a 508c PDF for CARTS. There are various changes in here that address issues that were found using the accessibility checker in adobe acrobat.

## How to test

This can be tested by viewing a state form, clicking the print button, and then entire form. And then clicking the print button on the new tab that gets opened. After receiving the generated PDF it can then be ran against the accessibility checker for adobe acrobat.

## Code authors checklist

- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests. [Testing Doc](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research)
- [ ] Necessary analytics were added, or no new analytics were required
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned the PR to myself

## Reviewers Checklist (Two different people)

- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

## Assignee

- [ ] I have closed the PR after the review and necessary changes (squashing preferred)

### CODE REVIEW

File: frontend/react/src/components/fields/CMSLegend.js [hunk 1]
```diff
@@ -18,23 +18,36 @@ const CMSLegend = ({ hideNumber, hint, id, label }) => {
...
-      {!hideNumber && labelBits}
-      {label}
-      {hint && (
-        <div className="ds-c-field__hint">
-          <Text>{hint}</Text>
-        </div>
-      )}
-    </legend>
-  );
+  console.log({ questionType });
```
Comment (benjaminpaige) [inline frontend/react/src/components/fields/CMSLegend.js:21]: can prob remove this

File: frontend/react/src/components/fields/Checkbox.js [hunk 2]
```diff
@@ -17,18 +16,35 @@ const Checkbox = ({ onChange, question, ...props }) => {
...
-    >
-      {label}
-    </Choice>
-  ));
+  const radioButttonList = question.answer.options.map(
+    ({ label, value: checkBoxValue }, idx) => {
+      return (
+        <div
+          style={{ margin: "2rem 0" }}
+          key={props.name + "-" + value + "-" + idx}
```
Comment (benjaminpaige) [inline frontend/react/src/components/fields/Checkbox.js:24]: we could prob interp this if we wanted to clean it up. no biggie either way

File: frontend/react/src/components/fields/Question.js [hunk 3]
```diff
@@ -107,6 +107,7 @@ const Question = ({
   // Check if question should be shown based on pathname
   const pageDisable = showQuestionByPath(window.location.pathname);
 
+  console.log({ question });
```
Comment (benjaminpaige) [inline frontend/react/src/components/fields/Question.js:110]: .

File: frontend/react/src/components/fields/Radio.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React, { useState } from "react";
 import PropTypes from "prop-types";
-import { Choice } from "@cmsgov/design-system-core";
+// import { Choice } from "@cmsgov/design-system-core";
```
Comment (benjaminpaige) [inline frontend/react/src/components/fields/Radio.js:3]: .

File: frontend/react/src/components/layout/Part.js [hunk 5]
```diff
@@ -59,10 +59,10 @@ const Part = ({
   return (
     <div id={partId}>
       {title && (
-        <h4 className="h4-pdf-bookmark">
+        <h3 className="h4-pdf-bookmark">
```
Comment (benjaminpaige) [inline frontend/react/src/components/layout/Part.js:62]: maybe update the classname?

File: frontend/react/src/components/layout/Section.js [hunk 6]
```diff
@@ -13,16 +13,16 @@ const selectSectionNumber = (sectionId) => {
...
     <div className="section-basic-info ds-l-col--9 content">
       <div className="main">
         <PageInfo />
-        <h2 className="print-only">
+        {/* <h2 className="print-only">
           {sectionId !== 0 && <span>Section {sectionId}: </span>}
           {title}
-        </h2>
-        <h2 className="screen-only">{title}</h2>
+        </h2> */}
```
Comment (benjaminpaige) [inline frontend/react/src/components/layout/Section.js:24]: .

File: frontend/react/src/components/fields/CMSLegend.js [hunk 7]
```diff
@@ -18,23 +18,36 @@ const CMSLegend = ({ hideNumber, hint, id, label }) => {
...
+    return (
+      <p>
+        {!hideNumber && labelBits}
+        {!questionType.includes("text") && label}
+        {hint && (
+          <div className="ds-c-field__hint">
+            <Text>{hint}</Text>
+          </div>
+        )}
+      </p>
```
Comment (benjaminpaige) [inline frontend/react/src/components/fields/CMSLegend.js:39]:
```suggestion
      <div>
        {!hideNumber && <p>{labelBits}</p>}
        {!questionType.includes("text") && <p>{label}</p>}
        {hint && (
          <div className="ds-c-field__hint">
            <Text>{hint}</Text>
          </div>
        )}
      </div>
```

maybe?

File: frontend/react/src/components/fields/Checkbox.js (hunk 2 above)
Comment (TheDanMiller) [reply frontend/react/src/components/fields/Checkbox.js:24]: image
It looks lie code climate picks it up in two places as repeated code

File: frontend/react/src/components/fields/Checkbox.js [hunk 8]
```diff
@@ -17,22 +16,37 @@ const Checkbox = ({ onChange, question, ...props }) => {
...
-  ));
+  const radioButttonList = question.answer.options.map(
+    ({ label, value: checkBoxValue }, idx) => {
+      return (
+        <div
+          className="radio-container"
+          key={props.name + "-" + value + "-" + idx}
+        >
+          <input
+            id={props.name + "-" + value}
```
Comment (benjaminpaige) [inline frontend/react/src/components/fields/Checkbox.js:27]:
```suggestion
            id={`${props.name} -${value}`}
```

**Review by benjaminpaige — APPROVED** [review]

File: frontend/react/src/components/fields/Radio.js [hunk 9]
```diff
@@ -20,32 +19,40 @@ const Radio = ({ onChange, onClick, question, ...props }) => {
...
+    <div className="radio-container" key={props.name + "-" + value}>
+      <input
+        key={value}
+        checked={checked === value}
+        type="radio"
+        value={value}
+        {...childProps}
+        {...props}
+        onChange={onCheck}
+        onClick={unCheck}
```
Comment (BearHanded) [inline frontend/react/src/components/fields/Radio.js:40]: Do we need to be firing both these onChange and onClicks in an input component? It looks like it clears and sets its contents back to back.

File: frontend/react/src/components/fields/SynthesizedTable.js [hunk 10]
```diff
@@ -3,10 +3,15 @@ import PropTypes from "prop-types";
 import { connect } from "react-redux";
 import synthesizeValue from "../../util/synthesize";
 
-const SynthesizedTable = ({ headers, rows }) => {
+const SynthesizedTable = ({ headers, rows, question }) => {
+  console.log({ question });
```
Comment (BearHanded) [inline frontend/react/src/components/fields/SynthesizedTable.js:7]: Remove log

File: frontend/react/src/components/layout/UploadComponent.js [hunk 11]
```diff
@@ -20,6 +20,7 @@ class UploadComponent extends Component {
   }
 
   componentDidMount = async () => {
+    console.log({ question: this.props.question });
```
Comment (BearHanded) [inline frontend/react/src/components/layout/UploadComponent.js:23]: Remove log

**Review by BearHanded — CHANGES_REQUESTED** [review]

File: frontend/react/src/components/fields/Radio.js (hunk 9 above)
Comment (13bfrancis) [reply frontend/react/src/components/fields/Radio.js:40]: I'm not sure. In looking at master it looks like this same logic existed so maybe we can leave it for now. I agree though, that it does seem strange.

**Review by BearHanded — APPROVED** [review]

**Pull request merged**